### ISSUE TITLE
it just stops after a few days

### ISSUE THREAD

**Issue by tkameroski** [TomWhitwell/SlowMovie#25]
I keep encountering a problem with the player freezing after a few days (usually around 4 days) of running.  It happens no matter which version of the code I use (master or 0.2, or missionfloyd’s python3 version.)  It happens when running on a Pi zero and on a Pi 4b.   No error is displayed on the attached monitor.  The frame count stops and it stops displaying new frames.  After reboot it works well again for a while.  Anyone else having that problem?  Any ideas what is going wrong?

**Comment by TomWhitwell** [comment]
There was an issue like that discussed here - https://github.com/TomWhitwell/SlowMovie/issues/18 - but shouldn't be in all those versions

**Comment by missionfloyd** [comment]
If it's the issue mentioned above, it's fixed in the latest waveshare driver (https://github.com/waveshare/e-Paper). Try installing that.

**Comment by tkameroski** [comment]
I installed the updates driver.  Will give it a try.  thanks

**Comment by flyingangel007** [comment]
Sorry for my question, but how can I change the driver? Is it enough to overwrite the "/home/pi/SlowMovie/e-paper/RaspberryPi&JetsonNano/python/lib/waveshare_epd/epd7in5_V2.EPD" file with the new version?

Thanks for your help!

**Comment by flyingangel007** [comment]
> The waveshare driver (https://github.com/waveshare/e-Paper) was updated a few days ago. Try using that.

Sorry, but I think the driver is an old version - just uploaded new. The Version is 4.0 from 2019-06-20.

**Comment by missionfloyd** [comment]
> Sorry for my question, but how can I change the driver? Is it enough to overwrite the "/home/pi/SlowMovie/e-paper/RaspberryPi&JetsonNano/python/lib/waveshare_epd/epd7in5_V2.EPD" file with the new version?
> 
> Thanks for your help!

To install: in the `e-paper/RaspberryPi_JetsonNano/python` folder there's a file called `setup.py`. Run `sudo python setup.py install`
To uninstall: `sudo pip uninstall waveshare_epd`

Or, you can just put the `waveshare_epd` folder in `/home/pi/SlowMovie`

**Comment by tkameroski** [comment]
Still stopping after several days.  The pi keeps working, the program just stops showing new frames.  No error shown.  Works after reboot for a bunch more days then same thing happens.  I have tried several versions of SlowMovie or three different versions of pi (zero, 3 and 4b) using two different 7in5_V2 e-paper displays.  Used the driver from v0.2 here and the most recent one from Waveshare github and same on all from fresh reinstalled sd cards.  The longest it has worked has been 6 days before no new refreshes.  No one else encountering this problem?  Any ideas how to track down where the driver may be entering an unending loop or other potential explanations?  Is there a way to have the pi reboot if the program doesn't update the log file with a new frame number after a specified amount of time if the problem cannot be identified or corrected?  thanks for your help.

**Comment by TomWhitwell** [comment]
Very strange - definitely not having that problem here - mine runs for months quite happily

**Comment by flyingangel007** [comment]
> Still stopping after several days. The pi keeps working, the program just stops showing new frames. ..

Hi!

I have the exact same problem. But just on one of my frames - the second one is at my mothers home - and hers is working perfect. I'll try to put her SD-card in my frame in a few days, so I can check if its a software or hardware problem.

**Comment by qubist** [comment]
@victorwisc do you have the logs from the new code? `tail -f /home/pi/SlowMovie/slowmovie.log`

**Comment by victorwisc** [comment]
I do but there is not much in there.

[2021-04-13 21:28:42,014] Update interval: 120
[2021-04-13 21:28:42,014] Frame increment: 4
[2021-04-13 21:28:43,572] Resuming at frame 38192
[2021-04-13 21:28:43,573] Playing 'Lifeboat.mp4'
[2021-04-13 21:28:43,574] Video info: 140166 frames, 23.976fps, duration: 5846.08s
[2021-04-19 22:17:32,763] Exiting Program

I did do another git pull before restarting this time and I notice my log is now different, with INFO on the line so I did not have the most recent but should have been at least when it was restarted on 4/13.

**Comment by qubist** [comment]
Yeah, should be recent enough. It looks like something killed the program and the code on [line 33](https://github.com/TomWhitwell/SlowMovie/blob/main/slowmovie.py#L33) was run. Were you running it as a service? @missionfloyd might know more also

**Comment by victorwisc** [comment]
Yea, I am guessing the Exiting Program was when I restarted the Pi. It is running as a service. It is strange that it only happens every 4 or 5 days so far. I was considering simply restarting the Pi every 3 days in the middle of the night and I would probably never know it. :)

**Comment by flyingangel007** [comment]
Thanks for the new version and perfect readme and step-by-step manual! My frame is now working since a view days without any problems. My frame is not running as a service.

**Comment by tkameroski** [comment]
Mine has been running for 15 days on a Pi4 without stopping.  That is by far the longest so fingers are crossed the problem is resolved.  I followed the directions posted at that time exactly as written and used the repo that was available at noon that day.  The only other thing I did differently than in my previous attempts was I used it as a hat instead of the SPI cables to connect to the pi.  Unfortunately there doesn't appear to be a log file for this one so if it crashes I wont be able to look back to see what may have happened.

I have a second e-paper connected to a Pi Zero I started yesterday.  In the past the Zero crashes within 24 hours but it is still running.  I used the installer for that one.  I have this logging at the debug level.

Both are running as a service.

Will post if they stop.

Victorwise: I tried an auto-reboot at 3 AM every night a few weeks ago.  The reboots went well but the Pi4 images still stopped after 4-5 days.  Can't figure out why that would happen.  Hopefully another scheduled reboot attempt won't be necessary if things keep running well.

**Comment by qubist** [comment]
Seems like something about the newer updates fixed whatever issue this was! I'm going to close the issue, but please feel free to update or reopen if the issue pops up again!

**Comment by Siriusbizzy** [comment]
I recently made two new sets (latest version via the automated install) with Raspberry Pi Zero's and 7in_5 Waveshare displays (driver has been updated) connected one via the supplied paper-HAT and one with the wired cable. I used the installer and both run as a service just fine...for around 45-50 minutes and then the screens on both just stop updating while the program still seems to be running. Tried the same SD card with a RP3 and it ran for at least two hours before also halting screen updates. I'm kind of out of ideas, any suggestions where to start digging? Apart from what has already been stated here? The log file just states the program started and is running. (maybe it is just Pulp Fiction being a b#tch? :p)
